PVC/polygon2voxel.m

```diff
@@ -0,0 +1,168 @@
+%   Mode : (optional) if set to:
+%               'none', The vertices data is directly used as coordinates
+%                       in the voxel volume.
+%               'wrap', The vertices data is directly used as coordinates
+%                       in the voxel volume, coordinates outside are 
+%                       circular wrapped to the inside.
+%               'auto', The vertices data is translated and 
+%                       scaled with a scalar to fit inside the new volume.
+%               'center', coordinate 0,0,0 is set as the center of the volume
+%                       instead of the corner of the voxel volume.
+%               'clamp', The vertices data is directly used as coordinates
+%                       in the voxel volume, coordinates outside are 
+%                       clamped to the inside.
+%   (Optional)
+%   Yxz : If true (default) use Matlab convention 1th dimension Y, 
+%         2th dimension X, and last dimension Z. Otherwise 1th 
+%         dimension is X, 2th Y and last Z.
+%
+%                        
+% Outputs,
+%   Volume : The 3D logical volume, with all voxels part of the discretized
+%           mesh one, and all other voxels zero.
+%
+% Example,
+%   % Compile the c-coded function
+%   mex polygon2voxel_double.c -v
+%
+%   % Load a triangulated mesh of a sphere
+%   load sphere; 
+%
+%   % Show the mesh
+%   figure, patch(FV,'FaceColor',[1 0 0]); axis square;
+%
+%   % Convert the mesh to a voxelvolume
+%   Volume=polygon2voxel(FV,[50 50 50],'auto');
+%
+%   % Show x,y,z slices
+%   figure,
+%   subplot(1,3,1), imshow(squeeze(Volume(25,:,:)));
+%   subplot(1,3,2), imshow(squeeze(Volume(:,25,:)));
+%   subplot(1,3,3), imshow(squeeze(Volume(:,:,25)));
+%
+%   %  Show iso surface of result
+%   figure, patch(isosurface(Volume,0.1), 'Facecolor', [1 0 0]);
+%
+% Example2,
+%   % Compile the c-coded function
+%   mex polygon2voxel_double.c -v
+%
+%   % Make A Volume with a few blocks
+%   I = false(120,120,120);
+%   I(40:60,50:70,60:80)=1; I(60:90,45:75,60:90)=1;
+%   I(20:60,40:80,20:60)=1; I(60:110,35:85,10:60)=1;
+%
+%   % Convert the volume to a triangulated mesh
+%   FV = isosurface(I,0.8);
+%
+%   % Convert the triangulated mesh back to a surface in a volume
+%   J = polygon2voxel(FV,[120, 120, 120],'none'); 
+%   % Fill the volume
+%   J=imfill(J,'holes');
+% 
+%   % Differences between original and reconstructed
+%   VD = abs(J-I);
+%
+%   % Show the original Mesh and Mesh of new volume
+%   figure, 
+%   subplot(1,3,1),  title('original')
+%     patch(FV,'facecolor',[1 0 0],'edgecolor','none'), camlight;view(3);
+%   subplot(1,3,2), title('converted');
+%     patch(isosurface(J,0.8),'facecolor',[0 0 1],'edgecolor','none'), camlight;view(3);
+%   subplot(1,3,3), title('difference');
+%     patch(isosurface(VD,0.8),'facecolor',[0 0 1],'edgecolor','none'), camlight;view(3); 
+%
+% Function is written by D.Kroon University of Twente (May 2009)

```

Comment:
@gmangeat did you check if we have the permission to add in our repos with MIT license?